torch/__init__.py

```diff
@@ -1335,15 +1335,19 @@ def __init__(self, mode, options, dynamic):
             ), "triton.cudagraphs does not support dynamic shapes"
 
     def apply_mode(self, mode: Optional[str]):
-        if mode is None:
-            return
-        elif mode == "default":
+        if mode is None or mode == "default":
             pass
         elif mode == "reduce-overhead":
-            self.config["triton.cudagraphs"] = True
+            self.apply_options({
+                "triton.cudagraphs": True,
+                "size_asserts": False,
```

Comment:
I think if we're using cudagraphs size_asserts wouldn't have any affect anyways?

Reply:
It helps if cudagraph gets disabled (due to mutation, etc) and on CPU.